RPI/src/processor.py

```diff
@@ -33,58 +33,139 @@ def __init__(self, stringer, protocol_handler, expectation_handler):
         self.__protocol_handler = protocol_handler
         self.__expectation_handler = expectation_handler
 
-    def process(self, input):
+    def process(self, input_):
         """
         Returns output after processing passed input.
-        @param input  The input to be processed
+        @param input_  The input to be processed
         """
 
-        if input == b"Ping":
-            return self.__ping()
-        elif input == b"Motion Detected":
-            return self.__motion_detected()
-        elif input == b"White Detected":
-            return self.__color_detected(0)
-        elif input == b"Black Detected":
-            return self.__color_detected(1)
-        else:
-            return [b"Unknown Message"]
+        try:
+            # Main interactions:
+            if input_ == "Ping":
+                return self.__ping()
+            elif input_ == "Motion Detected":
+                return self.__motion_detected()
+            elif input_ == "Primary White":
+                return self.__primary_color_detected(0)
+            elif input_ == "Primary Black":
+                return self.__primary_color_detected(1)
+            elif input_ == "Secondary White":
+                return self.__secondary_color_detected(0)
+            elif input_ == "Secondary Black":
+                return self.__secondary_color_detected(1)
+            elif input_ == "Confirm Blocker Extended":
+                return self.__blocker_extended()
+            elif input_ == "Confirm Blocker Retracted":
+                return self.__blocker_retracted()
+            elif input_ == "Confirm Pusher Push":
+                return self.__pusher_pushed()
+            elif input_ == "Confirm Stringer Push":
+                return self.__disk_stringed()
+
+            # Startup interactions:
+            elif input_ == "White Set":
+                pass    # TODO implement
+            elif input_ == "Black Set":
+                pass    # TODO implement
+
+            # Error interactions:
+            elif input_ == "Error Occurred":
+                pass    # TODO implement
+            else:
+                return ["Unknown Message"]
+        except Exception:
+            # TODO add error handling code
+            pass
 
     def __ping(self):
         """Returns output in case of ping."""
 
         self.__expectation_handler.ping()
+        self.__protocol_handler.inform_alive()
         expired_outputs = self.__expectation_handler.get_expired_outputs()
-
-        if expired_outputs:
-            return expired_outputs
-        else:
-            return [b"Pong"]
+        return ["Pong"] + expired_outputs
 
     def __motion_detected(self):
         """Returns output in case of motion."""
 
         if not self.__stringer.is_complete():
             # The stringer still needs disks
-            self.__expectation_handler.add(b"Confirm Extend Blocker", [b"Retract Blocker"], 10)
-            return [b"Extend Blocker", b"Scan Color"]
```

Comment:
Why do we no longer scan color when motion is detected? Also, there will probably have to be two separate methods, just like color, for primary and secondary motion.

Reply:
We scan color after the blocker has been extended.

Reply:
![communicator_to_processor](https://user-images.githubusercontent.com/64268431/111658269-7a6ed100-880c-11eb-85a1-2a9a9bb04022.png)

Reply:
Since everything is synchronous, we should never have two outgoing messages at the same time.

Reply:
Stringing the disk is a separate process, so it could be that we have one outgoing message for each process.